metaseq/checkpoint_utils.py

```diff
@@ -316,7 +320,8 @@ def _is_checkpoint_sharded(checkpoint_files) -> bool:
             "--model-parallel. If you are working on a new script, it may also mean "
             "you failed to fsdp_wrap or you have an unnecessary fsdp_wrap."
         )
-    sd = torch_load_cpu(checkpoint_files[0])
+    # We don't need the optimizer state here, since we're just loading the model for the config.
+    sd = load_and_pop_last_optimizer_state(checkpoint_files[0])
```

Comment:
Dumb q - is there a particular reason for popping optimizer state in a "is checkpoint sharded" function rather than something more inference-specific? 

(I get that we don't need the optimizer state while loading the model for generation, but `checkpoint_utils.py` is also used to load models for training, yeah?)

Reply:
Yeah, that's a good point... We use the is_sharded information to identify how the model is to be loaded, and we need to see the config for doing that. And the only way to load the config right now is to load the whole .pt file... (Which we were doing earlier using torch_load_cpu.

The change I made just drops the optimizer state after loading to reduce some memory overhead.